ev3dev/display.py

```diff
@@ -267,3 +277,77 @@ def update(self):
             self.mmap[:] = self._img_to_rgb565_bytes()
         else:
             raise Exception("Not supported")
+
+    def image_filename(self, filename, clear_screen=True, x1=0, y1=0, x2=None, y2=None):
+
+        if clear_screen:
+            self.clear()
+
+        filename_im = Image.open(filename)
+
+        if x2 is not None and y2 is not None:
+            return self._img.paste(filename_im, (x1, y1, x2, y2))
+        else:
+            return self._img.paste(filename_im, (x1, y1))
+
+    def shapes_line(self, clear_screen=True, x1=0, y1=0, x2=0, y2=0, line_color='black', width=1):
+
+        if clear_screen:
+            self.clear()
+
+        return self.draw.line((x1, y1, x2, y2), fill=line_color, width=width)
+
+    def shapes_circle(self, clear_screen=True, x=50, y=50, radius=40, fill_color='black', outline_color='black'):
+
+        if clear_screen:
+            self.clear()
+
+        x1 = x - radius
+        y1 = y - radius
+        x2 = x + radius
+        y2 = y + radius
+
+        return self.draw.ellipse((x1, y1, x2, y2), fill=fill_color, outline=outline_color)
+
+    def shapes_rectangle(self, clear_screen=True, x=0, y=0, width=0, height=0, fill_color='black', outline_color='black'):
+
+        if clear_screen:
+            self.clear()
+
+        return self.draw.rectangle((x, y, width, height), fill=fill_color, outline=outline_color)
+
+    def shapes_point(self, clear_screen=True, x=0, y=0, point_color='black'):
+
+        if clear_screen:
+            self.clear()
+
+        return self.draw.point((x, y), fill=point_color)
+
+    def text_pixels(self, text, clear_screen=True, x=0, y=0, text_color='black', font=None):
+
+        if clear_screen:
+            self.clear()
+
+        if font is not None:
+            assert font in fonts.available(), "%s is an invalid font" % font
+            return self.draw.text((x, y), text, fill=text_color, font=fonts.load(font))
+        else:
+            return self.draw.text((x, y), text, fill=text_color)
+
+    def text_grid(self, text, clear_screen=True, x=0, y=0, text_color='black', font=None):
```

Comment:
Document what these two text methods do... I'm a bit confused myself

Reply:
k

Reply:
Isn't there a page somewhere that shows what all of the different fonts look like on the EV3 screen?  I'm looking but can't find it...was going to include that in the doc string

Reply:
http://python-ev3dev.readthedocs.io/en/latest/other.html#ev3dev.fonts.load (https://github.com/rhempel/ev3dev-lang-python/commit/e6a1ddef12819e599355261a1fd53c5616c5cdf2)

I couldn't find info on how this was generated... somewhere I think @ddemidov had a Jupyter notebook with some code for it.

Reply:
found it
http://ev3dev-lang.readthedocs.io/projects/python-ev3dev/en/stable/other.html#bitmap-fonts

Reply:
#225 links to the notebook with the code used to generate the image.